presto-hive/src/main/java/io/prestosql/plugin/hive/HiveConfig.java

```diff
@@ -672,6 +673,19 @@ public HiveConfig setTranslateHiveViews(boolean translateHiveViews)
         return this;
     }
 
+    public boolean isHideDeltaLakeTables()
+    {
+        return hideDeltaLakeTables;
+    }
+
+    @Config("hive.hide-delta-tables")
```

Comment:
Add validation that if `hive.hide-delta-tables` is true then Glue metastore is in use.

I wanted to suggest a snippet like below but it does not work as we do not have metastore type in `HiveConfig`. You will figure sth out :p

```java
    @AssertTrue(message = "hive.hide-delta-tables works only with Glue metastore")
    public boolean isDeltaLakeHidingCompatibleWithMetastoreType()
    {
        return !hideDeltaLakeTables || hiveMetastoreIsGlue
    }
```

Reply:
Oh I see you added validation to metastore code.

Reply:
Yes, I want this to be covered by thrift too. Otherwise I'd add this to Glue...Config only